tools/SourceKit/lib/SwiftLang/SwiftIndexing.cpp

```diff
@@ -367,3 +368,81 @@ void SwiftLangSupport::indexSource(StringRef InputFile,
   SKIndexDataConsumer IdxDataConsumer(IdxConsumer);
   index::indexSourceFile(CI.getPrimarySourceFile(), IdxDataConsumer);
 }
+
+static void emitIndexDataForSourceFile(SourceFile &PrimarySourceFile,
+                                       StringRef IndexStorePath,
+                                       StringRef IndexUnitOutputPath,
+                                       const CompilerInstance &Instance) {
+  const auto &Invocation = Instance.getInvocation();
+  const auto &Opts = Invocation.getFrontendOptions();
+
+  bool isDebugCompilation;
+  switch (Invocation.getSILOptions().OptMode) {
+    case OptimizationMode::NotSet:
+    case OptimizationMode::NoOptimization:
+      isDebugCompilation = true;
+      break;
+    case OptimizationMode::ForSpeed:
+    case OptimizationMode::ForSize:
+      isDebugCompilation = false;
+      break;
+  }
+
+  (void) index::indexAndRecord(&PrimarySourceFile, IndexUnitOutputPath,
+                               IndexStorePath,
+                               !Opts.IndexIgnoreClangModules,
+                               Opts.IndexSystemModules,
+                               Opts.IndexIgnoreStdlib,
+                               Opts.IndexIncludeLocals,
+                               isDebugCompilation,
+                               Invocation.getTargetTriple(),
+                               *Instance.getDependencyTracker(),
+                               Invocation.getIRGenOptions().FilePrefixMap);
+}
+
+void SwiftLangSupport::indexToStore(
+    StringRef InputFile, ArrayRef<const char *> Args,
```

Comment:
Could we name this `PrimaryFilePath` to match the other requests? `InputFile` is fairly confusing as some requests also take the buffer name now.

Reply:
Done.